### PR TITLE
[Bexley][WW] Handle container removal requests

### PR DESCRIPTION
**Pull request by nephila-nacrea**
For https://github.com/mysociety/societyworks/issues/4576 and some of https://github.com/mysociety/societyworks/issues/4575.

The normal 'request new' journey also includes options for removal, if property has containers that can be removed.
There is also a separate journey that handles removals only.

[skip changelog]

### CODE REVIEW

File: templates/web/bexley/waste/summary_request.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <dd class="govuk-summary-list__value">[% data.request_reason %]</dd>
+  </div>
+
+  [% # Deliveries %]
+  [% IF requests.0.size %]
+    <div class="govuk-summary-list__row">
+      <dt class="govuk-summary-list__key govuk-summary-list__key--sub"><b>Deliver new or replacement:</b></dt>
+    </div>
+    [% FOR container IN requests.0 %]
+      [% INCLUDE container_row container = container %]
```
Comment (dracos) [inline templates/web/bexley/waste/summary_request.html:30]: You don't need  the `container = container`

File: perllib/FixMyStreet/App/Controller/Waste.pm [hunk 2]
```diff
@@ -775,30 +772,37 @@ sub construct_bin_request_form {
...
     my ($self, $c) = @_;
-    my $field_list = construct_bin_request_form($c);
 
-    $c->stash->{first_page} = 'request';
-    my $next = $c->cobrand->call_hook('waste_request_form_first_next');
-    my $title = $c->cobrand->call_hook('waste_request_form_first_title') || 'Which containers do you need?';
+    my %form_settings
+        = $c->cobrand->call_hook( 'construct_bin_request_form', $c );
+
+    my $field_list
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Waste.pm:779]: I think this four-line line is more easily parsable as `my $field_list = $form_settings{field_list} || construct_bin_request_form($c);` :)

File: perllib/FixMyStreet/Cobrand/Bexley/Waste.pm [hunk 3]
```diff
@@ -1366,22 +1485,14 @@ sub waste_request_form_update_field_list {
...
-    my $self = shift;
-
-    # Above-shop properties get a single default reason, so no need to send
-    # them to reason selection
-    return $self->{c}->stash->{property}{above_shop}
-        ? 'about_you'
-        : 'request_reason';
-}
-
 sub waste_munge_request_form_pages {
```
Comment (dracos) [inline perllib/FixMyStreet/Cobrand/Bexley/Waste.pm:1487]: Is this function needed any more? You could set first_page depending upon household_size_check in your main `construct_bin_request_form` above.

File: perllib/FixMyStreet/Cobrand/Bexley/Waste.pm [hunk 4]
```diff
@@ -1404,44 +1515,88 @@ sub waste_munge_request_data {
...
 
-        if ( $id_spaced eq ( $dc->{service_item_name} // '' ) ) {
-            $service = $dc;
+            last if $service;
         }
 
-        last if $service;
-    }
+        $data->{title}    = "Request removal of $service->{name}";
+        $data->{category} = $data->{category_removal};
```
Comment (dracos) [inline perllib/FixMyStreet/Cobrand/Bexley/Waste.pm:1597]: This block is almost identical to the request block just above if we want to factor those together

**Review by dracos — CHANGES_REQUESTED** [review]
Nice and clear, just a few small suggestions.
We can probably (later) look at if all the request forms can be refactored in a way to share bits (e.g. Kingston also has a removal page).

File: templates/web/bexley/waste/summary_request.html (hunk 1 above)
Comment (nephila-nacrea) [reply templates/web/bexley/waste/summary_request.html:30]: Yeah, this was probably me trying to make things work before I realised that passing-by-reference was what was causing me issues.

**Review by dracos — APPROVED** [review]
Looks good :)

**Pull request merged**